### PR TITLE
feat(client): add local filesystem scan in client/server mode

### LINKED ISSUE #634: trivy scan local filesystem not supported in client mode
trivy should support scan local filesystem using client mode
THX

### PR DESCRIPTION
**Pull request by afdesk**
## Description
Added a new option `--filesystem` to `client` command for support local filesystem scan in client/server mode.

## Related issues
- Close #634 

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've followed the [conventions](https://github.com/aquasecurity/trivy/blob/main/CONTRIBUTING.md#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/commands/client/option.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ type Option struct {
 	customHeaders []string
 	// this field is populated in Init()
 	CustomHeaders http.Header
+	FileSystem    bool
```
Comment (AndreyLevchenko) [inline pkg/commands/client/option.go:35]: could you add a verb to `bool`variable name

**Comment by AndreyLevchenko** [comment]
@afdesk  @knqyf263 
Given the 
https://github.com/aquasecurity/fanal/blob/main/artifact/local/fs.go#L139
what is advantage of using shared cache?

File: pkg/commands/client/option.go (hunk 1 above)
Comment (afdesk) [reply pkg/commands/client/option.go]: updated.

**Comment by NitroCao** [comment]
Can we also add support for `repo` subcommand in client/server mode?

**Comment by afdesk** [comment]
> Can we also add support for `repo` subcommand in client/server mode?

@NitroCao we'll take a look at `repo` command too. thanks

**Comment by afdesk** [comment]
this PR was replaced by #1829

**Pull request closed without merging**